src/test/java/teammates/test/cases/newaction/OriginCheckFilterTest.java

```diff
@@ -0,0 +1,130 @@
+package teammates.test.cases.newaction;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.testng.annotations.Test;
+
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+import teammates.common.util.StringHelper;
+import teammates.test.cases.BaseTestCase;
+import teammates.test.driver.MockFilterChain;
+import teammates.test.driver.MockHttpServletRequest;
+import teammates.test.driver.MockHttpServletResponse;
+import teammates.ui.newcontroller.OriginCheckFilter;
+
+/**
+ * SUT: {@link OriginCheckFilter}.
+ */
+public class OriginCheckFilterTest extends BaseTestCase {
+
+    private static final OriginCheckFilter FILTER = new OriginCheckFilter();
+
+    private MockHttpServletRequest mockRequest;
+    private MockHttpServletResponse mockResponse;
+    private MockFilterChain mockFilterChain;
+
+    private void setupMocks(String method) {
+        mockRequest = new MockHttpServletRequest(method, "http://localhost:8080");
+        mockRequest.setRequestedSessionId("requestedsessionid.node0");
+        mockResponse = new MockHttpServletResponse();
+        mockFilterChain = new MockFilterChain();
+    }
+
+    @Test
+    public void allTests() throws Exception {
```

Comment:
Separate test to `allTestsGetMethod` and `allTestPostMethod`?

Reply:
It does not really make sense to do that. The test cases are not really targeted at "GET" or "POST" per se but rather the different execution paths of the filter.